tests/org.dataflowanalysis.converter.tests/src/org/dataflowanalysis/converter/tests/TUHHPipeline.java

```
@@ -64,18 +98,65 @@ public void runPipeline() throws IOException {
+                .stream()
+                .allMatch(key -> datasets.stream()
+                        .anyMatch(path -> path.getFileName()
+                                .toString()
+                                .equals(key))));
+        assertTrue(CYCLIC_SINK_VARIANTS.keySet()
+                .stream()
+                .allMatch(key -> datasets.stream()
+                        .anyMatch(path -> path.getFileName()
+                                .toString()
+                                .equals(key))));
+
         for (var dataset : datasets) {
             assertTrue(Files.isDirectory(dataset.resolve("model_variants")));
             cleanTopLevelOfDataset(dataset);
             renameTxtVariants(dataset.resolve("model_variants"));
             moveTxtVariantsUp(dataset.resolve("model_variants"));
             convertTxtToJson(dataset);
             convertJsonToDFD(dataset);
+            filter(dataset);
         }
 
         Files.delete(converter);
     }
 
+    private void filter(Path dataset) throws IOException {
+        Set<Integer> variants = new HashSet<>(OUT_OF_SCOPE_VARIANTS);
+        var datasetName = dataset.getFileName()
+                .toString();
+        if (FAULTY_VARIANTS.containsKey(datasetName)) {
+            variants.addAll(FAULTY_VARIANTS.get(datasetName));
+        }
+        if (CYCLIC_SINK_VARIANTS.containsKey(datasetName)) {
+            variants.addAll(CYCLIC_SINK_VARIANTS.get(datasetName));
+        }
+
+        Files.list(dataset)
+                .forEach(path -> {
+                    if (Files.isRegularFile(path)) {
+                        for (var variant : variants) {
+                            if (path.getFileName()
+                                    .toString()
+                                    .contains("_" + variant + ".")) {
+                                try {
+                                    Files.delete(path);
+                                } catch (IOException e) {
```

I'm not sure whether this should fail silently